src/server/plugins/engine/components/NumberField.test.ts

```diff
@@ -190,7 +189,37 @@ describe('NumberField', () => {
         expect(viewModel.suffix).toEqual({ text: 'per item' })
       })
 
-      it('sets Nunjucks component step attribute', () => {
+      it('sets Nunjucks component inputmode attribute when precision is not defined', () => {
+        const componentCustom = new NumberField(
+          { ...def, schema: { precision: undefined } },
+          formModel
+        )
+
+        const viewModel = componentCustom.getViewModel(getFormData(99))
+
+        expect(viewModel.attributes).toEqual(
+          expect.objectContaining({
+            inputmode: 'numeric'
+          })
+        )
+      })
+
+      it('sets Nunjucks component inputmode attribute when precision is 0', () => {
+        const componentCustom = new NumberField(
+          { ...def, schema: { precision: 0 } },
+          formModel
+        )
+
+        const viewModel = componentCustom.getViewModel(getFormData(99))
+
+        expect(viewModel.attributes).toEqual(
+          expect.objectContaining({
+            inputmode: 'numeric'
+          })
+        )
+      })
+
+      it("Doesn't set Nunjucks component inputmode attribute when precision is positive", () => {
```

Comment:
Sorry, should probably find a Jest ESLint rule that keeps these lowercase

Cheers for all these tests

```suggestion
      it("does not set Nunjucks component inputmode attribute when precision is positive", () => {
```